nodebuilder/tests/tastora/api/cross_version_client_test.go

```diff
@@ -0,0 +1,392 @@
+	} else {
+		require.NoError(s.T(), err)
+	}
+	_, err = client.Share.GetNamespaceData(ctx, head.Height(), namespace)
+	if err != nil && strings.Contains(err.Error(), "data not available") {
+		s.T().Logf("Share.GetNamespaceData: data not available (known limitation with old light servers): %v", err)
+	} else {
+		require.NoError(s.T(), err)
+	}
+	_, err = client.Share.GetEDS(ctx, head.Height())
+	if err != nil && strings.Contains(err.Error(), "data not available") {
+		s.T().Logf("Share.GetEDS: data not available (known limitation with old light servers): %v", err)
+	} else {
+		require.NoError(s.T(), err)
+	}
+
+	if !skipGetRow {
+		rowCtx, rowCancel := context.WithTimeout(ctx, 120*time.Second)
+		_, err = client.Share.GetRow(rowCtx, head.Height(), 0)
+		rowCancel()
+		require.NoError(s.T(), err)
+	}
+
+	dasCtx, dasCancel := context.WithTimeout(ctx, 15*time.Second)
+	_, err = client.DAS.SamplingStats(dasCtx)
+	dasCancel()
+	if err != nil && !strings.Contains(err.Error(), "stubbed") && !strings.Contains(err.Error(), "deadline exceeded") {
+		require.NoError(s.T(), err)
+	}
+
+	blobCtx, blobCancel := context.WithTimeout(ctx, 30*time.Second)
+	_, err = client.Blob.GetAll(blobCtx, head.Height(), []share.Namespace{namespace})
+	blobCancel()
+	if err != nil && strings.Contains(err.Error(), "deadline exceeded") {
+		s.T().Logf("Skipping Blob.GetAll timeout (known limitation with old light servers): %v", err)
+	} else {
+		require.NoError(s.T(), err)
+	}
+
+	testBlob, err := nodeblob.NewBlobV0(namespace, []byte("cross-version test blob"))
+	if err == nil {
+		submitHeight, err := client.Blob.Submit(ctx, []*nodeblob.Blob{testBlob}, state.NewTxConfig())
+		if err == nil && submitHeight > 0 {
+			_, err = client.Blob.Get(ctx, submitHeight, namespace, testBlob.Commitment)
+			require.NoError(s.T(), err)
+
+			proof, err := client.Blob.GetProof(ctx, submitHeight, namespace, testBlob.Commitment)
+			require.NoError(s.T(), err)
+
+			_, err = client.Blob.Included(ctx, submitHeight, namespace, proof, testBlob.Commitment)
+			require.NoError(s.T(), err)
+		}
+	}
+}
+
+func (s *CrossVersionClientTestSuite) runCompatTest(ctx context.Context, clientVersion, serverRPCAddr string, skipGetRow bool) error {
+	dockerClient := s.f.GetDockerClient()
+	networkID := s.f.GetDockerNetwork()
+
+	imageName := fmt.Sprintf("ghcr.io/celestiaorg/celestia-node-compat-test:%s", clientVersion)
+
+	_, _, err := dockerClient.ImageInspectWithRaw(ctx, imageName)
+	if err != nil {
+		s.T().Logf("Pulling Docker image %s (this may take a while)...", imageName)
+		pullResp, pullErr := dockerClient.ImagePull(ctx, imageName, image.PullOptions{})
+		if pullErr != nil {
+			s.T().Logf("Failed to pull image %s: %v", imageName, pullErr)
+			s.T().Logf("Attempting to build image locally as fallback...")
+
+			if buildErr := s.buildCompatTestImageLocally(ctx, clientVersion, imageName); buildErr != nil {
```

Comment:
for compat tests, I would imagine we never actually want to build the image locally, it should always be something that has been built from a tag, or this PR (prior to this go test running, so as a previous step in the workflow)